desc/objectives/_coils.py

```diff
@@ -134,6 +139,11 @@ def _prune_coilset_tree(coilset):
             ValueError,
             "Only use toroidal resolution for coil grids.",
         )
+        errorif(
```

Comment:
I don't love this, can you instead do something like giving the CoilConvexity objective a `self._endpoint` attribute, and in this build check if hasattr(self,_endpoint) and usethat, else set it to False.

Then have this error check statement be after the `super.build()` statement inside of `CoilConvexity`'s build method